### PR TITLE
Fix new instances of unnecessary_this in latest SDK

### PR DESCRIPTION
**Pull request by kevmoo**
Fixed a number of prefer_final_locals outside of the example pkg

### CODE REVIEW

File: example/lib/example.dart [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Person {
 
   Person(this.firstName, this.lastName, this.dateOfBirth,
       {this.middleName, this.lastOrder, List<Order> orders})
-      : this.orders = orders ?? <Order>[];
+      : orders = orders ?? <Order>[];
```
Comment (jakemac53) [inline example/lib/example.dart]: hmmm I actually much prefer the `this` here for disambiguation purposes

Comment (jakemac53) [reply example/lib/example.dart]: :man_shrugging:

Comment (kevmoo) [reply example/lib/example.dart]: Yeah, this confused the heck out of me, too.

I guess if it's the value being assigned it **has** to be for `this` – but it is a bit weird.

Comment (jakemac53) [reply example/lib/example.dart]: Ya it is technically not necessary but I always have preferred it....

**Pull request merged**